src/pages/pupils/lessons/[lessonSlug]/index.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  GetStaticPathsResult,
+  GetStaticProps,
+  GetStaticPropsResult,
+  NextPage,
+} from "next";
+
+import getPageProps from "@/node-lib/getPageProps";
+import {
+  getFallbackBlockingConfig,
+  shouldSkipInitialBuild,
+} from "@/node-lib/isr";
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+import {
+  PupilExperienceView,
+  PupilExperienceViewProps,
+} from "@/components/PupilViews/PupilExperience";
+import { requestLessonResources } from "@/components/PupilComponents/pupilUtils/requestLessonResources";
+import { resolveOakHref } from "@/common-lib/urls";
+
+/**
+ * Test URLs:
+ *
+ * Non-published legacy lesson - should result in 404:
+ * http://localhost:3000/pupils/lessons/what-is-video-c4v68d
+ *
+ * Published legacy lesson:
+ * http://localhost:3000/pupils/lessons/myths-and-folktales-6cwk0c
+ *
+ * Published lesson:
+ * http://localhost:3000/pupils/lessons/transverse-waves
+ *
+ *
+ */
+
+const PupilsCanonicalPage: NextPage<PupilExperienceViewProps> = ({
+  curriculumData,
+  hasWorksheet,
+  backUrl,
+}) => {
+  return (
+    <PupilExperienceView
+      curriculumData={curriculumData}
+      hasWorksheet={hasWorksheet}
+      backUrl={backUrl}
+    />
+  );
+};
+
+type PupilCanonicalPageURLParams = {
+  lessonSlug: string;
+};
+
+export const getStaticPaths = async () => {
+  if (shouldSkipInitialBuild) {
+    return getFallbackBlockingConfig();
+  }
+
+  const config: GetStaticPathsResult<PupilCanonicalPageURLParams> = {
+    fallback: "blocking",
+    paths: [],
+  };
+  return config;
+};
+
+export const getStaticProps: GetStaticProps<
+  PupilExperienceViewProps,
+  PupilCanonicalPageURLParams
+> = async (context) => {
+  return getPageProps({
+    page: "pupils-lesson-overview-legacy-canonical::getStaticProps",
+    context,
+    getProps: async () => {
+      if (!context.params) {
+        throw new Error("No context.params");
+      }
+      const { lessonSlug } = context.params;
+
+      const curriculumData =
+        await curriculumApi2023.pupilLessonOverviewCanonical({
+          lessonSlug,
+        });
+
+      if (!curriculumData) {
+        return {
+          notFound: true,
+        };
+      }
+
+      const backUrl = curriculumData.isLegacy
+        ? `${resolveOakHref({
+            page: "classroom",
+          })}/units/${curriculumData.unitSlug}`
+        : null;
+
+      const { transcriptSentences, hasWorksheet } =
+        await requestLessonResources({ curriculumData });
+
+      const results: GetStaticPropsResult<PupilExperienceViewProps> = {
+        props: {
+          curriculumData: {
+            ...curriculumData,
+            transcriptSentences: transcriptSentences ?? [],
+          },
+          hasWorksheet,
+          backUrl,
+        },
+      };
+
+      return results;
```

Comment:
Just trying to understand the rationale behind all this. Why can't we redirect to the regular lesson page from here instead of creating a new page altogether 

`pupilLessonOverviewCanonical` gives us the canonical `unitSlug` and `programmeSlug` so that should be possible.

Is it down to the variance in the logic for the `backUrl` or an SEO thing?

Reply:
The lessons can exist in more than one place so there is not an appropriate place to redirect to . Ultimately there will be extra behaviours here in terms of navigation as there are in the Teacher Journey

Reply:
Probably the other legacy canonical page could have been forwarded to this one but it's only a temporary measure in anycase